provider/src/main/scala/com/actian/spark_vector/provider/RequestHandler.scala

```diff
@@ -92,17 +93,21 @@ class RequestHandler(sqlContext: SQLContext, val auth: ProviderAuth) extends Log
     writeJobResult(result)
   }
 
+  private def getMessage(cause: Throwable): String = cause.getMessage + (if (cause.getCause != null) s", caused by: ${getMessage(cause.getCause)}" else "")
+
   /** Handle the failure of a query request */
   private def handleFailure(cause: Throwable)(implicit socket: SocketChannel) = closeResourceAfterUse(socket) {
     val result = cause match {
       case JobException(e, job, part) => {
-        logInfo(s"Job tr_id:${job.transaction_id}, query_id:${job.query_id} failed for part ${part.part_id}. Reason: ${cause.getMessage}")
+        val message = getMessage(e)
+        logInfo(s"Job tr_id:${job.transaction_id}, query_id:${job.query_id} failed for part ${part.part_id}. Reason: $message")
```

Comment:
We use logInfo for the main message and logDebug for the stack trace since we want to avoid polluting the logs with stack traces when there can just be very simple user errors, e.g. trying to insert a null value into a non-nullable colum. We use logError only when we cannot parse the JSON request, which should indicate an internal problem.